cdap-ui/app/features/tracker/directives/top-entity-graph/top-entity-graph.js

```diff
@@ -15,121 +15,216 @@
-
-      /* X AXIS */
-      let xAxis = d3.svg.axis()
-        .scale(x)
-        .orient('bottom')
-        .innerTickSize(-height - margin.top)
-        .outerTickSize(0)
-        .tickPadding(10)
-        .ticks(5);
-
-      svg.append('g')
-        .attr('class', 'x axis')
-        .attr('transform', 'translate(0, ' + height + ')')
-        .call(xAxis);
-
-      // Removing first tick
-      svg.select('.x.axis .tick')
-        .attr('display', 'none');
-
-
-      /* Y AXIS */
-      let yAxis = d3.svg.axis()
-        .scale(y)
-        .orient('left')
-        .ticks(5);
-
-      svg.append('g')
-        .attr('class', 'y axis')
-        .call(yAxis);
-
-      /* BARS */
-      svg.selectAll('.bar')
-          .data(scope.model.results)
-        .enter().append('rect')
-          .attr('class', 'bar')
-          .attr('y', (d) => { return y(d.label); })
-          .attr('x', -3)
-          .attr('rx', 3)
-          .attr('ry', 3)
-          .attr('height', y.rangeBand())
-          .attr('width', (d) => { return Math.abs(x(d.value)) + 3; });
-
-      /* Adding Links */
-      let sidebarElem = angular.element(parentContainer[0]).find('div');
-
-      if (scope.type === 'apps') {
-        angular.forEach(scope.model.results, (result) => {
-          let link = angular.element('<a></a>')
-            .attr('class', 'entity-link')
-            .attr('ui-sref', 'apps.detail.overview.programs({ appId: "' + result.label + '"})')
-            .text(result.label);
-
-            // TODO: Add ellipsis and tooltip
-
-          let elem = $compile(link)(scope);
-          elem.css('top', y(result.label) + margin.top - 1 + (y.rangeBand()/2) + 'px');
-
-          sidebarElem.append(elem);
-        });
+      if (scope.type === 'applications') {
+        fetchTopApps();
+      } else if (scope.type === 'programs') {
+        fetchTopPrograms();
       }
 
-      // TODO: Add type === 'programs'
+      function fetchTopApps () {
+        let params = {
+          namespace: $state.params.namespace,
+          entity: scope.type,
+          scope: scope,
+          limit: 5
+        };
+
+        if(scope.start) {
+          params.start = scope.start;
+        }
+        if(scope.end) {
+          params.end = scope.end;
+        }
+
+        myTrackerApi.getTopEntities(params)
+          .$promise
+          .then((response) => {
+            scope.model = {
+              results: response
+            };
+            if(response.length >= 1) {
+              renderEntityGraph();
+            } else {
+              let metricContainer = d3.select(element[0]).select('.graph-container');
+              metricContainer.append('div')
+              .attr('class', 'well')
+              .append('p')
+              .text('Not enough data');
+            }
+          }, (err) => {
+            console.log('ERROR', err);
+          });
+      }
 
-    }
+      function fetchTopPrograms () {
+        let params = {
```

Comment:
Are we having any difference between `fetchTopPrograms` and `fetchTopApps`? Can we re-use as one function?